src/test/testType.ts

```diff
@@ -91,11 +101,20 @@ export class TestTypeImpl {
   private _hook(name: 'beforeEach' | 'afterEach' | 'beforeAll' | 'afterAll', location: Location, fn: Function) {
     const suite = currentlyLoadingFileSuite();
     if (!suite)
-      throw new Error(`${name} hook can only be called in a test file`);
+      throw errorWithLocation(location, `${name} hook can only be called in a test file`);
     suite._hooks.push({ type: name, fn, location });
   }
 
   private _modifier(type: 'skip' | 'fail' | 'fixme' | 'slow', location: Location, ...modifierArgs: [arg?: any | Function, description?: string]) {
+    if (typeof modifierArgs[1] === 'function') {
+      throw errorWithLocation(location, [
+        `It looks like you are calling test.${type}() to declare a test. Call it inside the test instead:`,
```

Comment:
can we add todo and tell them to use todo? Skip inside the test is a bad pattern because it runs all my custom fixtures.

Reply:
We can, but until we did it, I'd like to help our users.

Reply:
https://github.com/microsoft/playwright/pull/7755